src/renderer/components/dock/create-resource.tsx

```diff
@@ -21,6 +24,22 @@ interface Props {
 @observer
 export class CreateResource extends React.Component<Props> {
   @observable error = "";
+  @observable templates:GroupSelectOption<SelectOption>[] = [];
+
+  async componentDidMount(){
+    let badgetTemplates = await createResourceStore.getLensTemplates();
+
+    badgetTemplates = await createResourceStore.addUserTemplates(badgetTemplates);
```

Comment:
This seems backwards since it requires getting the lens templates and then badge templates.

I think it would be better to get both in tandem and then perform a merge.

Reply:
Unfortunately, I didn't catch what do you mean
BadgedGroupSelectOption - it is an extended GroupSelectOption class that allows storing labels and badges as separated values. 
Basically, I use it for getting info from create-resource.store.ts without including react libraries into that file, then I convert BadgedGroupSelectOption to GroupSelectOption where GroupSelectOption.label: Node is a compilation of BadgedGroupSelectOption.label:string and BadgedGroupSelectOption.badge:string (via renderSelectItem)

Reply:
I mean that the building up of `BadgedGroupSelectOption[]` from the user templates folder should be done seperately from the lens template folder, and then merged.

Reply:
Why? Merging logic is the most part of addUserTemplates, it is because we need to check built-in-template-folder absence before adding a new one in the userTemplatesGroups and marking this one by Badge (or marking all user files in this group if the folder exists) 
We can leave this logic and convert BadgedGroupSelectOption[] to GroupSelectOption[] twice but It looks like some excess to me.

Reply:
The reason why I am asking about this is that both of them read from the file system. It would be both easier to test and easier to review if those two actions were split.